xen/arch/x86/domain.c

```diff
@@ -649,7 +650,8 @@ int arch_domain_create(struct domain *d, unsigned int domcr_flags,
         if ( (rc = init_domain_irq_mapping(d)) != 0 )
             goto fail;
 
-        if ( (rc = iommu_domain_init(d, false)) != 0 )
+        use_iommu = !!(domcr_flags & DOMCRF_use_iommu);
+        if ( (rc = iommu_domain_init(d, use_iommu)) != 0 )
```

Comment:
I add new flag to common domctl path not in arch specific path. Therefore I process this flag in a same way for both arch. It is up to x86 use this flag or not.